commands/addons/attach.js

```diff
@@ -23,12 +25,8 @@ function * run (context, heroku) {
     )
   }
 
-  let attachment = yield createAttachment(app, context.flags.as, context.flags.confirm)
-    .catch((err) => {
-      if (!err.body || err.body.id !== 'confirmation_required') throw err
-      return cli.confirmApp(app, context.flags.confirm, err.body.message)
-        .then(() => createAttachment(app, context.flags.as, app))
-    })
+  let attachment = yield util.trapConfirmationRequired(context, (confirm) => createAttachment(app, context.flags.as, confirm))
```

Comment:
rather than use a closure that wraps the promise with a callback I think it would be simpler to just have a catch handler:

```js
let attachment = yield createAttachment(app, context.flags.as, confirm).catch(util.confirmationRequired(context))
```

That would clean up the util function a lot.